### PR TITLE
Pushing to add reviewer.

### PR DESCRIPTION
**Pull request by Jamesw2013**

### CODE REVIEW

File: assignments/array-methods.js [hunk 1]
```diff
@@ -55,29 +55,58 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
+let fullName = []
+runners.forEach(runner => fullName.push(runner.first_name + " " +runner.last_name));
```
Comment (Mister-Corn) [inline assignments/array-methods.js:59]: Alternatively, you can process the string like this:

```
`${runner.first_name} ${runner.last_name}`
```

Remember them backticks!

File: assignments/array-methods.js [hunk 2]
```diff
@@ -55,29 +55,58 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
...
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//find all runners from any given company
+let getCompany = [];
+function runnersCompany(company){
+    runners.filter(function(runner){
+        if(runner.company_name=== company){
+            getCompany.push(JSON.stringify(runner));
+        }
```
Comment (Mister-Corn) [inline assignments/array-methods.js:87]: `filter` returns a new array filtered from the original array based on the callback function. In your case, you're not doing that at all, but instead pushing elements into an array you made yourself outside. In that case, `forEach` is more appropriate.

THis is what I wrote in another review:

> This works similarly to map. The filter array method:
> 
> * Creates a new array
> * Calls the callback function on each element on the old array
>   * It expects a true/false from the callback function
>     * If true, it pushes the element into the new array
>     * If false, it leaves it alone
> * Returns the new array

In that case, there'd be no need to make a new array and push into it: filter would automatically do that.

In short, do it like you did it on line 69 of this file.

File: assignments/array-methods.js [hunk 3]
```diff
@@ -55,29 +55,58 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
...
+console.log(alphaEmails);
+
+// Problem 3
+//find a runner's ID #
+function runnersNum (first, last){
+    runners.forEach(function(runner){
+        if(runner.first_name === first && runner.last_name === last){
+            console.log("This is " + first +"'s" + " id#:", runner.id);
+            return runner.id;
+        }
```
Comment (Mister-Corn) [inline assignments/array-methods.js:108]: Well done on this! :+1:

File: assignments/callbacks.js [hunk 4]
```diff
@@ -27,29 +27,46 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
 
 function getLength(arr, cb) {
   // getLength passes the length of the array into the callback.
+  return cb(arr.length);
 }
 
+//Testing getLength
+getLength (items, function(param){console.log("getLength test:", param);});
```
Comment (Mister-Corn) [inline assignments/callbacks.js:34]: You consistently pass a callback function which logs to the console whatever value is passed into it. Custom `console.log` messsages aside, it is essentially the same function.

Remember, you can use functions like they're values, and save them to variables. You can then use that variable to refer to that function:

```js
function getLength(arr, cb) {
  // getLength passes the length of the array into the callback.
  return cb(arr.length, 'getLength: '); 
}

function consoleLog(input, message = '') {
    /*  Input the result the funcion will pass to the callback
     *  Message is an optional string to spice up your console.log
     */
    console.log(message, input);
}

getLength(items, consoleLog); // console.log -> 'getLength: 4'
```

This is just an example. This could be simpler, but I wanted to keep around the little intro messages into your console logs.

File: assignments/closure.js [hunk 5]
```diff
@@ -1,6 +1,13 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
+function multiply(x){
+  return function (y) {
+    return x * y; 
+  };
+}
```
Comment (Mister-Corn) [inline assignments/closure.js:7]: An elegant closure. Well done.

**Review by Mister-Corn — APPROVED** [review]
## Objectives

### Higher Order Functions and Callbacks

* [x] Complete the problems provided to you but skip over stretch problems until you are complete with every other JS file first.

### Array Methods

* [x] Complete the problems provided to you but skip over stretch problems until you are complete with every other JS file first.

* [x] Notice the last three problems are up to you to create and solve.  This is an awesome opportunity for you to push your critical thinking about array methods, have fun with it.

### Closures

* [x] Complete the problems provided to you but skip over stretch problems until you are complete with every other JS file first.

## Comments

Great work, James! 💯 Check out my comments, ask me questions if you have any, and keep up the good work!

Also, you serious about meeting up? Let's do it when I come up to the Bay!

File: assignments/closure.js (hunk 5 above)
Comment (Jamesw2013) [reply assignments/closure.js:7]: Yeah it be cool to meetup! Im about a hour away! Depending on when you come we can just plan it around one of the sfbayarea lambda meetups too.
